Gems/AtomLyIntegration/CommonFeatures/Code/Source/PostProcess/EditorModeFeedback/EditorEditorModeFeedbackSystemComponent.cpp

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <PostProcess/EditorModeFeedback/EditorEditorModeFeedbackSystemComponent.h>
+#include <AzCore/Component/ComponentBus.h>
+#include <AzCore/Component/TransformBus.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzToolsFramework/Viewport/ViewportMessages.h>
+#include <AzToolsFramework/FocusMode/FocusModeInterface.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <Atom/RPI.Public/MeshDrawPacket.h>
+#include <Atom/RPI.Public/Shader/ShaderResourceGroup.h>
+#include <Atom/RPI.Public/Model/ModelLodUtils.h>
+#include <Atom/RPI.Public/MeshDrawPacket.h>
+#include <Atom/RPI.Public/Model/Model.h>
+#include <Atom/RPI.Public/Scene.h>
+#include <Atom/RPI.Public/ViewportContext.h>
+#include <Atom/RPI.Public/ViewportContextBus.h>
+#include <Atom/RPI.Public/DynamicDraw/DynamicDrawInterface.h>
+#include <Atom/Utils/Utils.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        //! Builds the mesh draw packets for the specified model.
+        static AZStd::vector<RPI::MeshDrawPacket> BuildMeshDrawPackets(
+            const RPI::View* view,
+            const AZ::Transform& worldTM,
+            const Data::Asset<RPI::ModelAsset>& modelAsset,
+            const Data::Instance<RPI::Material>& material,
+            const Data::Instance<RPI::ShaderResourceGroup>& meshObjectSrg)
+        {
+            AZStd::vector<RPI::MeshDrawPacket> meshDrawPackets;
+            const RPI::Model& model = *RPI::Model::FindOrCreate(modelAsset);
+            const auto modelLodIndex = RPI::ModelLodUtils::SelectLod(view, worldTM, model);
+            const Data::Asset<RPI::ModelLodAsset>& modelLodAsset = modelAsset->GetLodAssets()[0];
+            RPI::ModelLod& modelLod = *RPI::ModelLod::FindOrCreate(modelLodAsset, modelAsset).get();
```

Comment:
This should probably be a Data::Instance<RPI::ModelLod>, then de-reference it when you pass it into the drawPacket constructor below. That way you're holding on to a refcount.

It may be unlikely, but if this were the first thing to create the ModelLod instance, then you'd end up with a Data::Instance<RPI::ModelLod> returned from FindOrCreate, but if you don't actually keep a copy of the DataInstance<> (intrusive_ptr), then I think the refcount will just immediately drop to 0 and the ModelLod instance will be released.

Reply:
Fixed 👍